### PR TITLE
Apim 3510 bug unique endpoint names

### PR DESCRIPTION
**Pull request by jourdiw**
## Issue

https://gravitee.atlassian.net/browse/APIM-3510

## Description

Endpoint group and endpoint names must be unique for V4 APIs

## Additional context

🏗️ Your changes can be tested here and will be available soon:
      Console: [https://pr.team-apim.gravitee.dev/6138/console](https://pr.team-apim.gravitee.dev/6138/console)
      Portal: [https://pr.team-apim.gravitee.dev/6138/portal](https://pr.team-apim.gravitee.dev/6138/portal)
      Management-api: [https://pr.team-apim.gravitee.dev/6138/api/management](https://pr.team-apim.gravitee.dev/6138/api/management)
      Gateway v4: [https://pr.team-apim.gravitee.dev/6138](https://pr.team-apim.gravitee.dev/6138)
      Gateway v3: [https://pr.gateway-v3.team-apim.gravitee.dev/6138](https://pr.gateway-v3.team-apim.gravitee.dev/6138)

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-jakycxtzyd.chromatic.com)

### CODE REVIEW

File: gravitee-apim-console-webui/src/management/api/endpoints-v4/endpoint-group/api-endpoint-group.component.ts [hunk 1]
```diff
@@ -92,31 +93,30 @@ export class ApiEndpointGroupComponent implements OnInit, OnDestroy {
...
         },
-        [
-          Validators.required,
-          Validators.pattern(/^[^:]*$/),
-          isUniqueAndDoesNotMatchDefaultValue(
-            this.api.endpointGroups.reduce((acc, group) => [...acc, group.name], []),
-            group?.name,
-          ),
-        ],
+        [Validators.required, Validators.pattern(/^[^:]*$/), isUniqueAndDoesNotMatchDefaultValue(existingNames, this.endpointGroup.name)],
```
Comment (loriepisicchio) [inline gravitee-apim-console-webui/src/management/api/endpoints-v4/endpoint-group/api-endpoint-group.component.ts]: Sugggestion: I think we should trim the value before validation, and also before saving.

Comment (jourdiw) [reply gravitee-apim-console-webui/src/management/api/endpoints-v4/endpoint-group/api-endpoint-group.component.ts]: It is trimmed within the validator automatically. It is trimmed in the update here: https://github.com/gravitee-io/gravitee-api-management/pull/6138/files#diff-241ad4aa7d72b0f9f20ceb6fb78060c2a19af14a74127f8cb1282aa73e926979R147

File: gravitee-apim-console-webui/src/management/api/endpoints-v4/endpoint-group/create/api-endpoint-group-create.component.spec.ts [hunk 2]
```diff
@@ -288,8 +288,8 @@ describe('ApiEndpointGroupCreateComponent', () => {
     describe('Endpoint group name validation', () => {
       const EXISTING_ENDPOINT_GROUP = fakeEndpointGroupV4({
         type: 'kafka',
-        name: 'Existing endpoint group',
-        endpoints: [fakeKafkaMessageEndpoint({ name: 'Existing endpoint' })],
+        name: 'Existing endpoint group ',
```
Comment (loriepisicchio) [inline gravitee-apim-console-webui/src/management/api/endpoints-v4/endpoint-group/create/api-endpoint-group-create.component.spec.ts:291]: Feels strange that the existing value is not trimmed. I would prefer reversing the test and adding a trailing space when setting the value with the harness

**Review by loriepisicchio — APPROVED** [review]

File: gravitee-apim-console-webui/src/management/api/endpoints-v4/endpoint-group/create/api-endpoint-group-create.component.spec.ts (hunk 2 above)
Comment (jourdiw) [reply gravitee-apim-console-webui/src/management/api/endpoints-v4/endpoint-group/create/api-endpoint-group-create.component.spec.ts:291]: I guess it's a legacy test then 😅 . I'll add some more tests to cover different scenarios 👍

File: gravitee-apim-console-webui/src/management/api/endpoints-v4/endpoint/api-endpoint.component.spec.ts [hunk 3]
```diff
@@ -100,80 +101,202 @@ describe('ApiEndpointComponent', () => {
...
+            //       name: 'default',
+            //       type: 'kafka',
+            //       weight: 1,
+            //       inheritConfiguration: false,
+            //       configuration: {
+            //         bootstrapServers: 'localhost:9092',
+            //       },
+            //     },
+            //   ],
+            // },
```
Comment (loriepisicchio) [inline gravitee-apim-console-webui/src/management/api/endpoints-v4/endpoint/api-endpoint.component.spec.ts]: 🗑️  ?

**Review by loriepisicchio — APPROVED** [review]

**Pull request merged**